syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.ni.hardware.grpc";
option java_outer_classname = "ServerUtilities";
option csharp_namespace = "NationalInstruments.Hardware.Grpc";

package ni.hardware.grpc;

service ServerUtilities {
  rpc Reserve (ReserveRequest) returns (ReserveResponse) {};
  rpc IsReservedByClient (IsReservedByClientRequest) returns (IsReservedByClientResponse) {};
  rpc Unreserve (UnreserveRequest) returns (UnreserveResponse) {};
}

message ViSession {
  oneof session {
      string session_name = 1;
      sfixed64 id = 2;
  }
}

message ReserveRequest {
  ViSession session = 1;
  string client_reserve_id = 2;
}

message ReserveResponse {
  enum ReserveStatus {
      NONE = 0;
      INVALID_SESSION = 1;
      RESERVED = 2;
  }
  ReserveStatus status = 1;
}

message IsReservedByClientRequest {
  string client_reserve_id = 1;
}

message IsReservedByClientResponse {
  bool is_reserved = 2;
}

message UnreserveRequest {
  string client_reserve_id = 1;
}

message UnreserveResponse {
  bool is_unreserved = 1;
}
